var/spack/repos/builtin/packages/neurodamus-model/package.py

```diff
@@ -201,6 +217,32 @@ def setup_run_environment(self, env):
 fi
 
 # run with nrnivmodl in path
-set -x
-'{nrnivmodl}' -incflags '{incflags} '"$2" -loadflags '{loadflags} '"$3" "$1"
+set -xe
+
+if [ ! -d "$1" ]; then
+    echo "Please provide a valid directory with mod files"
+    exit -1
+fi
+
+if [ -n "{nrnivmodlcore_call}" ]; then
+    rm -rf _core_mods
+    mkdir _core_mods
+    touch $1/neuron_only_mods.txt  # ensure exists
+    for f in $1/*.mod; do
+        if ! grep $(basename $f) $1/neuron_only_mods.txt; then
+            cp $f _core_mods/
+        fi
+    done
+    {nrnivmodlcore_call} _core_mods
+    libpath=$(dirname */libcorenrnmech*)
+    extra_loadflags="-L $libpath -lcorenrnmech"
```

Comment:
I tried running a sim with `CoreNEURON` using
```
module load neurodamus-neocortex py-neurodamus
build_neurodamus.sh modfull  # modfull has all the neocortex files and I create a neuron_only_mods.txt as well
srun -n 1 neurodamus BlueConfig
```
and setting the following:
```
export NRNMECH_LIB_PATH=$(pwd)/x86_64/libnrnmech.so
unset BGLIBPY_MOD_LIBRARY_PATH
```
to avoid using the same libraries as the installed neurodamus but I got the following:
```
[INFO] Loading MECH lib: /gpfs/bbp.cscs.ch/project/proj16/magkanar/cvf/neocortex/blueconfigs/quick-v6/x86_64/libnrnmech.so
dlopen failed -
libcorenrnmech.so: cannot open shared object file: No such file or directory
```
and
```
bash-4.2$ ldd x86_64/libnrnmech.so | grep core
        libcorenrnmech.so => not found
```
Maybe there is some issue with the linking of the `coreneuron` library, or I messed something up?
Running the sim with:
```
srun -n 40 ./x86_64/special -mpi -python $NEURODAMUS_PYTHON/init.py
```
worked fine.

Reply:
`./x86_64/special` works because `libcorenrnmech.so` is in the location of executable i.e. `./x86_64` and loader finds it without searching anywhere else.

I was suggesting to set rpath to `<pwd>/x86_64/`  so that this type error won't appear. cc: @ferdonline

Reply:
Hum... some more joy with relative rpaths, so people can at least relocate the folder... :)

Reply:
Btw, I can try to code the rpath, but IIRC rpath from libraries aren't necessarily respected.
The script shall write a note for the user to add LD_LIBRARY_PATH if he finds that error. Since we are not going for static linking I see that as the only solution.

Reply:
> but IIRC rpath from libraries aren't necessarily respected.

I didn't see / remember an issue with this because we use it in coreneuron. Unless something specific corner case you remember? So I think we can set rpath and it should work.

Reply:
I don't remember the details, but that why cmake puts all ld libraries and rpaths recursively in the final executable. there were differences in Mac, Linux ld and Gold. I will study the topic and make a presentation some day :)